app/controllers/concerns/authentication.rb

```diff
@@ -28,21 +34,25 @@ def current_user
   end
 
   def current_user_is_admin?
-    @current_user_is_admin ||= current_user_signed_in? && current_user.has_role?(:admin)
+    @current_user_is_admin ||= (
+      current_user_signed_in? && current_user.has_role?(:admin)
+    )
   end
 
   def current_user_signed_in?
     return true if current_user
   end
 
-  def verify_admin
-    redirect_to root_url unless current_user.has_role? :admin
+  def authenticate_user_or_redirect_to_root!
+    if !current_user
+      redirect_to(
+        root_url,
+        alert: 'You need to sign in for access to this page.'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.